### PR TITLE
SAK-40784 tomcat/lib commons-text

### PR DESCRIPTION
**Pull request by axxter99**

### CODE REVIEW

**Comment by mpellicer** [comment]
My two cents, will be good if we upgrade to 1.6 too...

**Comment by axxter99** [comment]
@ottenhoff SAK-40784:
dhorwitz@pollen:~/Sakai/tomcat> find . -name "commons-text*.jar"
./lib/commons-text-1.5.jar
dhorwitz@pollen:~/Sakai/tomcat>

**Comment by axxter99** [comment]
commons-text 1.6

**Comment by axxter99** [comment]
@ern & @ottenhoff ;

a) deploy/pom.xlm commons.text: <scope>compile</scope>, master/pom.xlm: <scope>provided</scope> (OK)
OR
b) deploy/pom.xlm commons.text: <scope>compile</scope>, master/pom.xlm: <scope>compile</scope>
OR
c) deploy/pom.xlm commons.text: <scope>compile</scope>, master/pom.xlm: <scope>provided</scope>, webapp & components (lessonbuilder, etc) pom.xlm commons-text: <scope>compile</scope>

?

**Comment by ern** [comment]
yep C!

**Comment by axxter99** [comment]
@ern OK

File: common/import-pack/pom.xml [hunk 1]
```diff
@@ -51,6 +51,11 @@
       <groupId>org.sakaiproject.common</groupId>
       <artifactId>sakai-archive-import</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.apache.commons</groupId>
+      <artifactId>commons-text</artifactId>
+      <scope>compile</scope>
```
Comment (ern) [inline common/import-pack/pom.xml]: remove the scope here as this should be inherited from master

File: lessonbuilder/tool/pom.xml [hunk 2]
```diff
@@ -102,6 +102,11 @@
             <artifactId>commons-io</artifactId>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-text</artifactId>
+            <scope>compile</scope>
```
Comment (ern) [inline lessonbuilder/tool/pom.xml]: remove the scope here as this should be inherited from master

File: samigo/samigo-app/pom.xml [hunk 3]
```diff
@@ -129,8 +129,13 @@
...
         <dependency>
-            <groupId>commons-lang</groupId>
-            <artifactId>commons-lang</artifactId>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
+        </dependency>
+       <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-text</artifactId>
+            <scope>compile</scope>
```
Comment (ern) [inline samigo/samigo-app/pom.xml]: remove the scope here as this should be inherited from master

File: samigo/samigo-pack/pom.xml [hunk 4]
```diff
@@ -31,6 +31,11 @@
 			<groupId>org.sakaiproject.kernel</groupId>
 			<artifactId>sakai-component-manager</artifactId>
 		</dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-text</artifactId>
+            <scope>compile</scope>
```
Comment (ern) [inline samigo/samigo-pack/pom.xml]: remove the scope here as this should be inherited from master

File: webservices/cxf/pom.xml [hunk 5]
```diff
@@ -192,6 +192,11 @@
             <artifactId>commons-lang3</artifactId>
             <scope>compile</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-text</artifactId>
+            <scope>compile</scope>
```
Comment (ern) [inline webservices/cxf/pom.xml]: remove the scope here as this should be inherited from master

File: assignment/impl/pom.xml [hunk 6]
```diff
@@ -95,6 +95,7 @@
         <dependency>
             <groupId>org.apache.commons</groupId>
             <artifactId>commons-text</artifactId>
+            <scope>compile</scope>
```
Comment (ern) [inline assignment/impl/pom.xml]: remove the scope here as this should be inherited from master

File: common/import-pack/pom.xml [hunk 7]
```diff
@@ -51,6 +51,11 @@
       <groupId>org.sakaiproject.common</groupId>
       <artifactId>sakai-archive-import</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.apache.commons</groupId>
+      <artifactId>commons-text</artifactId>
+      <scope>inherited</scope>
```
Comment (ern) [inline common/import-pack/pom.xml]: I don't think this is a valid scope

**Comment by steveswinsburg** [comment]
FYI a webapp can declare its own library and it will still work even if deployed to the common/shared tomcat classloader due to Tomcat's order of precendence when loading classes (webapps before common/shared). So removing them from apps is not required to get this done but is a good thing to do for consistency.

**Comment by ern** [comment]
> FYI a webapp can declare its own library and it will still work even if deployed to the common/shared tomcat classloader due to Tomcat's order of precendence when loading classes (webapps before common/shared). So removing them from apps is not required to get this done but is a good thing to do for consistency.

There is one caveat, when a class is loaded from the same lib but by different classloaders issues can arise because those classes are not equal when compared and inheritance also becomes an issue because isAssignableFrom will return false and ClassCast Exceptions will occur. This is one of primary reasons for webapps to use the libs in shared vs declaring their own. Only when a webapp is sure that an instance of a class will not be used outside its classloader is this safe to do.

**Comment by ern** [comment]
This PR still isn't correct as most of the scopes need to be removed as they shouldn't be overriding master.

**Pull request closed without merging**